Composer/packages/server/src/services/project.ts

```diff
@@ -159,6 +159,16 @@ export class BotProjectService {
     return projectId.toString();
   };
 
+  // clean project registry based on path to avoid reuseing the same id
+  public static cleanProject = async (location: LocationRef): Promise<void> => {
+    for (const key in BotProjectService.projectLocationMap) {
+      if (BotProjectService.projectLocationMap[key] === location.path) {
+        delete BotProjectService.projectLocationMap[key];
+      }
+    }
```

Comment:
```suggestion
    delete BotProjectService.projectLocationMap[location.path];
```

Reply:
not sure i understand this, the projectLocationMap is  id => path mapping, i need to remove by value, not by key, make sense?

Reply:
That makes sense.